compiler/rustc_ast/src/mut_visit.rs

```diff
@@ -1076,7 +1084,15 @@ pub fn noop_visit_item_kind<T: MutVisitor>(kind: &mut ItemKind, vis: &mut T) {
             vis.visit_ty(self_ty);
             items.flat_map_in_place(|item| vis.flat_map_impl_item(item));
         }
-        ItemKind::Trait(box Trait { unsafety, is_auto: _, generics, bounds, items }) => {
+        ItemKind::Trait(box Trait {
+            impl_restriction,
+            unsafety,
+            is_auto: _,
+            generics,
+            bounds,
+            items,
+        }) => {
+            noop_visit_restriction(impl_restriction, vis);
```

Comment:
Should this be `visit_restriction` instead?

Reply:
Honestly, I'm not sure why there are the many methods that are called `noop_*`, since they _do_ visit the various items. I simply copied the same style for other visitors.

Reply:
Looking everything over myself, I'll be changing it to `vis.visit_restriction(impl_restriction)`.

Reply:
Are we looking at the same thing? https://github.com/rust-lang/rust/pull/106074/files#diff-77aad7466ba75ca9ed918b5c2b6a1953d69951b6b4b7b92f7a94b289c40dfb6bR1111

Reply:
Hm, I guess not. Not really sure what I meant when I've written this 3 weeks ago. nvm then.